### ISSUE TITLE
MATCH query returns incorrect result set when an index is created

### ISSUE THREAD

**Issue by InverseIntegral** [neo4j/neo4j#12887]
The following
```
CREATE (:L {p:"test"})
MATCH (n:L) WHERE n.p STARTS WITH lTrim(n.p) RETURN COUNT(n)
```
returns
```
╒══════════╕
│"COUNT(n)"│
╞══════════╡
│1         │
└──────────┘
```

When I then run the following:
```
CREATE INDEX FOR (n:L) ON (n.p)
MATCH (n:L) WHERE n.p STARTS WITH lTrim(n.p) RETURN COUNT(n)
```
the result becomes
```
╒══════════╕
│"COUNT(n)"│
╞══════════╡
│0         │
└──────────┘
```

I would expect the result to still be 1.

**Neo4j Version:** 4.4.6
**Operating System:** 5.17.6 Arch Linux
**API:** Java

### Expected behavior
I would expect the second result to be the same as the first one.

### Actual behavior
The result bcomes 0.

**Comment by InverseIntegral** [comment]
I did some more testing and this seems to only happen if `lTrim` appears in the boolean expression.

**Comment by HannesSandberg** [comment]
Thanks, we will investigate and come back to you!

**Comment by InverseIntegral** [comment]
This seems to be fixed now in newer versions (confirmed on 4.4.10 enterprise).

**Comment by sherfert** [comment]
This has been fixed in 4.3.14, 4.4.8 and 5.0.0
